tt_metal/hw/inc/debug/dprint_tensix.h

```diff
@@ -5,76 +5,171 @@
     dbg_halt();
     MATH({
         // Determine the format of the data in the destination register
         uint32_t data_format_reg_field_value = READ_HW_CFG_0_REG_FIELD(ALU_FORMAT_SPEC_REG2_Dstacc);
 
-        #ifndef ARCH_GRAYSKULL
-        //ALU_ACC_CTRL_Fp32 does not exist for GS
+#ifndef ARCH_GRAYSKULL
+        // ALU_ACC_CTRL_Fp32 does not exist for GS
         if (READ_HW_CFG_0_REG_FIELD(ALU_ACC_CTRL_Fp32_enabled)) {
-            data_format_reg_field_value = 0; // Override the data format to tt::DataFormat::Float32
+            data_format_reg_field_value =
+                (uint32_t)DataFormat::Float32;  // Override the data format to tt::DataFormat::Float32
         }
-        #endif
+#endif
 
+        bool is_float32 = data_format_reg_field_value == (uint32_t)DataFormat::Float32;
+        bool is_swizzled = false;
+        bool is_remapped = false;
+
+#ifdef ARCH_BLACKHOLE
+        is_remapped = READ_HW_CFG_0_REG_FIELD(DEST_ACCESS_CFG_remap_addrs) == 1;
+        is_swizzled = READ_HW_CFG_0_REG_FIELD(DEST_ACCESS_CFG_swizzle_32b) == 1;
+#endif
         // Print the contents
-        DPRINT << FIXED() << SETPRECISION(2);
-        uint32_t rd_data[8+1]; // data + array_type
+        DPRINT << FIXED() << SETPRECISION(PRECISION);
         DPRINT << "Tile ID = " << tile_id << ENDL();
 
-        // print faces 0 & 1
-        int face_r_dim = 16;
-        for (int row = 0; row < face_r_dim; row++) {
-            // face 0
-            dbg_read_dest_acc_row(row + 64 * tile_id, rd_data);
-            DPRINT << SETW(6) << " " << TYPED_U32_ARRAY(TypedU32_ARRAY_Format_Tensix_Config_Register_Data_Format_Type, data_format_reg_field_value, rd_data, 8);
-        }
-        if constexpr (print_by_face) {
-            DPRINT << ENDL();
-        }
-
-        for (int row = 0; row < face_r_dim; row++) {
-            // face 1
-            dbg_read_dest_acc_row(row + face_r_dim + 64 * tile_id, rd_data);
-            DPRINT << SETW(6) << " " << TYPED_U32_ARRAY(TypedU32_ARRAY_Format_Tensix_Config_Register_Data_Format_Type, data_format_reg_field_value, rd_data, 8) << ENDL();
-        }
-
-        for (int row = 0; row < face_r_dim; row++) {
-            // face 2
-            dbg_read_dest_acc_row(row + 2*face_r_dim + 64 * tile_id, rd_data);
-            DPRINT << SETW(6) << " " << TYPED_U32_ARRAY(TypedU32_ARRAY_Format_Tensix_Config_Register_Data_Format_Type, data_format_reg_field_value, rd_data, 8);
-        }
-
-        if constexpr (print_by_face) {
-            DPRINT << ENDL();
-        }
-
-        for (int row = 0; row < face_r_dim; row++) {
-            // face 3
-            dbg_read_dest_acc_row(row + 3*face_r_dim + 64 * tile_id, rd_data);
-            DPRINT << SETW(6) << " " << TYPED_U32_ARRAY(TypedU32_ARRAY_Format_Tensix_Config_Register_Data_Format_Type, data_format_reg_field_value, rd_data, 8) << ENDL();
+        for (int face_id = 0; face_id < 4; ++face_id) {
+            for (int row_id = 0; row_id < 16; ++row_id) {
```

Comment:
Are the constants 4, 16 just part of our architecture? Could we have defines/constexprs for these?

Reply:
Added constants.